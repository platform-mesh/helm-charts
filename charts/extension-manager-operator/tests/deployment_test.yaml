suite: operator
templates:
  - deployment.yaml
  - deployment-server.yaml
  - service-account.yaml
  - cluster-role.yaml
  - cluster-rolebinding.yaml
values:
  - ../test-values.yaml
chart:
  version: 1.0.0
  appVersion: 1.0.0
release:
  name: extension-manager-operator
tests:
  - it: operator match the snapshot
    asserts:
      - matchSnapshot: {}
    set:
      istio:
        enabledOverride: true
      health:
        portOverride: 8081
        liveness:
          pathOverride: /readyz
      metrics:
        port: 8080
      deployment:
        resources:
          limits:
            cpuOverride: 260m
            memoryOverride: 512Mi
          requests:
            cpuOverride: 40m
            memoryOverride: 50Mi
  - it: operator match the snapshot with tracing
    templates:
      - deployment.yaml
      - deployment-server.yaml
    set:
      tracing:
        enabled: true
        collector:
          endpoint: test:1234
      metrics:
        port: 8080
    asserts:
      - matchSnapshot: {}
  - it: deployment with external kubeconfig
    template: deployment.yaml
    set:
      kcp:
        enabled: true
      kubeconfigSecret: "test-kubeconfig"
      extraVolumeMounts:
        - name: test
          mountPath: /test
      extraVolumes:
        - name: test
          secret:
            secretName: test
    asserts:
      - matchSnapshot: {}
  - it: has all health checks
    template: deployment.yaml
    asserts:
      - equal:
          path: spec.template.spec.containers[0].livenessProbe.httpGet.path
          value: /healthz
      - equal:
          path: spec.template.spec.containers[0].readinessProbe.httpGet.path
          value: /readyz
      - equal:
          path: spec.template.spec.containers[0].startupProbe.httpGet.path
          value: /readyz
