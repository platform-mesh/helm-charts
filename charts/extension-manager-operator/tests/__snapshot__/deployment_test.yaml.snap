deployment with external kubeconfig:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator-operator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "false"
          labels:
            app: extension-manager-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - operator
                - --kcp-enabled
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/extension-manager-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/platform-mesh/extension-manager-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: extension-manager-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts:
                - mountPath: /test
                  name: test
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - name: test
              secret:
                secretName: test
enable Gateway API:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: extension-manager-operator
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - contentconfigurations
          - contentconfigurations/status
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - coordination.k8s.io
        resources:
          - leases
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - get
          - list
          - watch
          - create
          - update
          - patch
          - delete
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: extension-manager-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: extension-manager-operator
    subjects:
      - kind: ServiceAccount
        name: extension-manager-operator
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator-server
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "false"
          labels:
            app: extension-manager-operator
            module: server
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - server
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/extension-manager-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/platform-mesh/extension-manager-operator:1.0.0
              imagePullPolicy: IfNotPresent
              name: extension-manager-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /readyz
                  port: 8088
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 3
                httpGet:
                  path: /readyz
                  port: 8088
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
  4: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator-operator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "false"
          labels:
            app: extension-manager-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/extension-manager-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/platform-mesh/extension-manager-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: extension-manager-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
          volumes: null
  5: |
    apiVersion: gateway.networking.k8s.io/v1
    kind: HTTPRoute
    metadata:
      name: extension-manager-operator-server
    spec:
      parentRefs:
        - group: gateway.networking.k8s.io
          kind: Gateway
          name: k8sapi-gateway
          sectionName: websecure
      rules:
        - backendRefs:
            - group: ""
              kind: Service
              name: extension-manager-operator-server
              port: 8088
          matches:
            - path:
                type: PathPrefix
                value: /custom-validate
  6: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: extension-manager-operator
      namespace: NAMESPACE
operator match the snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: extension-manager-operator
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - contentconfigurations
          - contentconfigurations/status
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - coordination.k8s.io
        resources:
          - leases
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - get
          - list
          - watch
          - create
          - update
          - patch
          - delete
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: extension-manager-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: extension-manager-operator
    subjects:
      - kind: ServiceAccount
        name: extension-manager-operator
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator-server
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
          labels:
            app: extension-manager-operator
            module: server
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - server
                - --leader-elect
                - --metrics-bind-address=:8080
                - --health-probe-bind-address=:8081
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/extension-manager-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/platform-mesh/extension-manager-operator:1.0.0
              imagePullPolicy: IfNotPresent
              name: extension-manager-operator
              ports:
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /readyz
                  port: 8088
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 3
                httpGet:
                  path: /readyz
                  port: 8088
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
  4: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator-operator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
          labels:
            app: extension-manager-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - operator
                - --leader-elect
                - --metrics-bind-address=:8080
                - --health-probe-bind-address=:8081
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/extension-manager-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/platform-mesh/extension-manager-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              name: extension-manager-operator
              ports:
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
          volumes: null
  5: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: extension-manager-operator
      namespace: NAMESPACE
  6: |
    apiVersion: networking.istio.io/v1alpha3
    kind: VirtualService
    metadata:
      name: extension-manager-operator-server
      namespace: NAMESPACE
    spec:
      gateways:
        - NAMESPACE/gateway
      hosts:
        - '*'
      http:
        - match:
            - uri:
                prefix: /validate
          route:
            - destination:
                host: extension-manager-operator-server.NAMESPACE.svc.cluster.local
                port:
                  number: 8088
operator match the snapshot with tracing:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator-server
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "false"
          labels:
            app: extension-manager-operator
            module: server
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - server
                - --leader-elect
                - --metrics-bind-address=:8080
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/extension-manager-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
                - --tracing-enabled=true
                - --tracing-config-service-name=extension-manager-operator-server.NAMESPACE
                - --tracing-config-service-version=1.0.0
                - --tracing-config-collector-endpoint=test:1234
              env: null
              image: ghcr.io/platform-mesh/extension-manager-operator:1.0.0
              imagePullPolicy: IfNotPresent
              name: extension-manager-operator
              ports:
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /readyz
                  port: 8088
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 3
                httpGet:
                  path: /readyz
                  port: 8088
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator-operator
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "false"
          labels:
            app: extension-manager-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - operator
                - --leader-elect
                - --metrics-bind-address=:8080
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/platform-mesh/extension-manager-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
                - --tracing-enabled=true
                - --tracing-config-service-name=extension-manager-operator.NAMESPACE
                - --tracing-config-service-version=1.0.0
                - --tracing-config-collector-endpoint=test:1234
              env: null
              image: ghcr.io/platform-mesh/extension-manager-operator:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: extension-manager-operator
              ports:
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
          volumes: null
